src/DiffEqGPU.jl

```diff
@@ -21,6 +21,45 @@ using Random
 using Setfield
 using ForwardDiff
 
+"""
+Wrapper for modifying parameters to contain additional data. Useful for simulating
+trajectories with different time span values.
+"""
+struct ParamWrapper{P, T}
+    params::P
+    data::T
+end
+
+function Adapt.adapt_structure(to, ps::ParamWrapper{P, T}) where {P, T}
+    ParamWrapper(adapt(to, ps.params),
+                 adapt(to, ps.data))
+end
+
+@kernel function gpu_kernel(f, du, @Const(u),
+                            @Const(params::AbstractArray{ParamWrapper{P, T}}),
+                            @Const(t)) where {P, T}
+    i = @index(Global, Linear)
+    @inbounds p = params[i].params
+    @inbounds tspan = params[i].data
+    @views @inbounds f(du[:, i], u[:, i], p, t)
+    @inbounds for j in 1:size(du, 1)
+        du[j, i] = du[j, i] * (tspan[2] - tspan[1])
+    end
+end
+
+@kernel function gpu_kernel_oop(f, du, @Const(u),
+                                @Const(params::AbstractArray{ParamWrapper{P, T}}),
+                                @Const(t)) where {P, T}
+    i = @index(Global, Linear)
+    @inbounds p = params[i].params
+    @inbounds tspan = params[i].data
+
+    @views @inbounds x = f(u[:, i], p, t)
+    @inbounds for j in 1:size(du, 1)
+        du[j, i] = x[j] * (tspan[2] - tspan[1])
```

Comment:
no handling of the dt?

Reply:
No, not yet. I'll need to think about the logic concerning how to pass different dt's and separate different time-spans. I will follow-up in another PR.